tools/toolchain/scripts/stage6/install_deepmd.sh

```diff
@@ -0,0 +1,115 @@
+#!/bin/bash -e
+
+# TODO: Review and if possible fix shellcheck errors.
+# shellcheck disable=all
+
+[ "${BASH_SOURCE[0]}" ] && SCRIPT_NAME="${BASH_SOURCE[0]}" || SCRIPT_NAME=$0
+SCRIPT_DIR="$(cd "$(dirname "$SCRIPT_NAME")/.." && pwd -P)"
+
+deepmd_ver="2.2.7"
+deepmd_sha256="605bbb0c3bcc847ecbfe7326bac9ff4cac5690accc8083122e3735290bb923ae"
+
+# shellcheck source=/dev/null
+source "${SCRIPT_DIR}"/common_vars.sh
+source "${SCRIPT_DIR}"/tool_kit.sh
+source "${SCRIPT_DIR}"/signal_trap.sh
+source "${INSTALLDIR}"/toolchain.conf
+source "${INSTALLDIR}"/toolchain.env
+
+[ -f "${BUILDDIR}/setup_deepmd" ] && rm "${BUILDDIR}/setup_deepmd"
+
+DEEPMD_LDFLAGS=''
+DEEPMD_LIBS=''
+DEEPMD_CXXFLAGS='-std=gnu++11 '
+
+! [ -d "${BUILDDIR}" ] && mkdir -p "${BUILDDIR}"
+cd "${BUILDDIR}"
+
+case "$with_deepmd" in
+  __INSTALL__)
+    echo "==================== Installing DeePMD ===================="
+    pkg_install_dir="${INSTALLDIR}/libdeepmd_c-${deepmd_ver}"
+    install_lock_file="${pkg_install_dir}/install_successful"
+    deepmd_root="${pkg_install_dir}"
+    if verify_checksums "${install_lock_file}"; then
+      echo "libdeepmd_c-${deepmd_ver} is already installed, skipping it."
+    else
+      if [ -f libdeepmd_c-${deepmd_ver}.tar.gz ]; then
+        echo "libdeepmd_c-${deepmd_ver}.tar.gz is found"
+      else
+        #download_pkg ${DOWNLOADER_FLAGS} -o libdeepmd_c-${deepmd_ver}.tar.gz ${deepmd_sha256} \
+        #https://github.com/deepmodeling/deepmd-kit/releases/download/v${deepmd_ver}/libdeepmd_c.tar.gz
+        wget ${DOWNLOADER_FLAGS} --quiet \
+          https://github.com/deepmodeling/deepmd-kit/releases/download/v${deepmd_ver}/libdeepmd_c.tar.gz
+        mv libdeepmd_c.tar.gz libdeepmd_c-${deepmd_ver}.tar.gz
+      fi
+      [ -d libdeepmd_c ] && rm -rf libdeepmd_c
+      echo "Installing from scratch into ${pkg_install_dir}"
+      tar -xzf libdeepmd_c-${deepmd_ver}.tar.gz
+      mv libdeepmd_c ${pkg_install_dir}
+      write_checksums "${install_lock_file}" "${SCRIPT_DIR}/stage6/$(basename ${SCRIPT_NAME})"
+    fi
+    DEEPMD_DFLAGS="-D__DEEPMD"
+    DEEPMD_CFLAGS="-I'${pkg_install_dir}/include'"
+    if [ "$DEEPMD_MODE" == "cpu" ]; then
+      DEEPMD_LDFLAGS="-L'${pkg_install_dir}/lib' -Wl,--no-as-needed -ldeepmd_op -ldeepmd_c -Wl,-rpath='${pkg_install_dir}/lib'"
+    elif [ "$DEEPMD_MODE" == "cuda" ]; then
+      DEEPMD_LDFLAGS="-L'${pkg_install_dir}/lib' -Wl,--no-as-needed -ldeepmd_op -ldeepmd_c -ldeepmd_dyn_cudart -ldeepmd_op_cuda -Wl,-rpath='${pkg_install_dir}/lib'"
+    fi
```

Comment:
This may not work as expected. `libdeepmd_op` is dynamically loaded during runtime. Linking or not linking it, or its dependency, will not change anything.

If the CUDA library is not found or the GPU is unavailable, the program will fall back to CPU mode. There is no need to have this option.